src/solver/optimisation/opt_appel_solveur_lineaire.cpp

```diff
@@ -79,7 +79,9 @@ static void fillModelerComponents(
   Modeler::Data* modelerData,
   OptimEntityContainer& optimEntityContainer)
 {
-    for (const auto& component: modelerData->system->Components())
+    const auto& components = modelerData->system->Components();
+    optimEntityContainer.reserveOptimComponents(components.size());
```

Comment:
Merge `addFromSystemComponent` and `reserveOptimComponents` into a single method, adding all components at once.
```suggestion
    optimEntityContainer.addFromSystemComponents(components);
```

Reply:
Done in the [next PR](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/3120)